testsuites/syncgateway/upgrade/test_upgrade.py

```diff
@@ -350,17 +393,35 @@ def update_docs(client, ls_url, ls_db, added_docs, auth, terminator_doc_id):
 
     docs_per_update = 3
     doc_revs = {}
+    deleted_docs = []
+
+    user_docs_to_delete = len(current_user_doc_ids) // 10
+    if user_docs_to_delete <= 0:
+        # Delete at least 1 doc
+        user_docs_to_delete = 1
+
+    user_docs_subset_to_delete = current_user_doc_ids[:user_docs_to_delete]
+    current_user_doc_ids = list(set(current_user_doc_ids) - set(user_docs_subset_to_delete))
+
+    log_info("Will delete {} docs".format(len(user_docs_subset_to_delete)))
+    log_info("user_docs_subset_to_delete: {}".format(user_docs_subset_to_delete))
+    for doc in user_docs_subset_to_delete:
+        user_docs_to_delete = client.get_doc(url=ls_url, db=ls_db, doc_id=doc, auth=auth)
+        log_info("Deleting doc_id: {}".format(doc))
+        client.delete_doc(url=ls_url, db=ls_db, doc_id=doc, rev=user_docs_to_delete['_rev'], auth=auth)
+        deleted_docs.append(doc)
 
     while True:
```

Comment:
I do not see any condition to come out of this loop other than Try/except. If there is a bug and does not throw HTTPError, then it might go to infinite loop. Better to have counter